### DISCUSSION TITLE
porting JPA entities to microstream

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#531] [Q&A]
Hi;
in my application I have about 100 JPA entities
I intend to take a small chunk and port to microstream.
i will use this thread to ask things that come up.
currently I have 2 questions:
1. all named queries in JPA should now be simply convenience methods using the java Streams API? I have a lot of named queries on each entity...
![entity](https://user-images.githubusercontent.com/2852437/217514811-6f134625-3d1e-4fb6-a379-596d194f0015.png)

2. all our DB tables have a primary key with autoincrement. if I undersand correctly, assuming I still want to have such a functionality in microstream, I need to implement this autoincrement feature myself whithin the app?
![entity-id](https://user-images.githubusercontent.com/2852437/217515204-614ab2b2-767f-4c4c-9dab-39c1f61b78d5.png)

thanks!

**Comment by hg-ms** [comment]
> all named queries in JPA should now be simply convenience methods using the java Streams API? I have a lot of named queries on each entity...

You are right, you need to replace the JPA queries by custom java methods.
To do so you can utilize the java steams API but you are not bound to that.


>all our DB tables have a primary key with autoincrement. if I undersand correctly, assuming I still want to have such a functionality in microstream, I need to implement this autoincrement feature myself whithin the app?

That’s correct too.

**Comment by guykatz** [comment]
thanks @hg-ms 
I think microstream should add a porting guide of sorts...  this is probably one of the first things for 30-40% of potentianl users...
I will use this thread to ask other things related to this port as things come up..
